### PR TITLE
Include internal fields into CosmosDB RetrievedDocument model

### PR DESCRIPTION
**Pull request by balanza**
This PR is to discuss the type modelling on `CosmosdbModel` and `CosmosdbModelVersioned` classes. 

The point is to have retrieved models to extend `CosmosResource` (formerly: `ResourceT`) instead of just `BaseModel`. This will require retrieved models' instances to explicitly handle the following fields:
```
_etag: t.string,
_rid: t.string,
_self: t.string,
_ts: t.number
```

### CODE REVIEW

File: src/utils/conversions.ts [hunk 1]
```diff
@@ -1,50 +0,0 @@
-import { NonEmptyString } from "italia-ts-commons/lib/strings";
```
Comment (gunzip) [inline src/utils/conversions.ts:1]: why delete this?

File: src/utils/messages.ts [hunk 2]
```diff
@@ -37,10 +37,10 @@ export async function getChannelStatus(
   const errorOrMaybeStatus = await notificationStatusModel
     .findOneNotificationStatusByNotificationChannel(notificationId, channel)
     .run();
-  return fromEither(errorOrMaybeStatus)
+  const c = fromEither(errorOrMaybeStatus)
```
Comment (gunzip) [inline src/utils/messages.ts:40]: why this change?

File: src/utils/cosmosdb_model.ts [hunk 3]
```diff
@@ -32,17 +32,17 @@ export type BaseModel = t.TypeOf<typeof BaseModel>;
 
 // An io-ts definition of Cosmos Resource runtime type
 // IDs are enforced to be non-empty string, as we're sure they are always valued when coming from db.
-export type ResourceT = t.TypeOf<typeof ResourceT>;
+export type CosmosResource = t.TypeOf<typeof CosmosResource>;
```
Comment (gunzip) [inline src/utils/cosmosdb_model.ts:35]: why renaming?

File: src/models/__tests__/user_data_processing.test.ts [hunk 4]
```diff
@@ -120,9 +124,7 @@ describe("createOrUpdateByNewOne", () => {
 
     const model = new UserDataProcessingModel(containerMock);
 
-    const result = await model
-      .createOrUpdateByNewOne(aUserDataProcessing)
-      .run();
+    const result = await model.upsert(aNewUserDataProcessing).run();
```
Comment (gunzip) [inline src/models/__tests__/user_data_processing.test.ts:127]: why this change here ? it looks like [createOrUpdateByNewOne](https://github.com/pagopa/io-functions-commons/blob/47a80c858b77fef036b27efd209c66b952e67440/src/models/user_data_processing.ts#L110) does more than a simple upsert

**Comment by cloudify** [comment]
This was actually my first design choice, bue later I've decided to not add this requirement to the Cosmos model as it may be left as a decision for each concrete implementation of the models - what benefit do you see in making this mandatory?

File: src/utils/conversions.ts (hunk 1 above)
Comment (balanza) [reply src/utils/conversions.ts:1]: I found this methods not to be used. Moreover, the linter now asks me to strip manual casting off, so what these methods are is actually just `as ModelId`

File: src/utils/messages.ts (hunk 2 above)
Comment (balanza) [reply src/utils/messages.ts:40]: debugging stuff, I remove it

**Comment by gunzip** [comment]
it looks like it needs rebase
